includes/gateway/class-omise-payment-googlepay.php

```diff
@@ -0,0 +1,237 @@
+<?php
+defined('ABSPATH') or die('No direct script access allowed.');
+
+class Omise_Payment_GooglePay extends Omise_Payment_Creditcard
+{
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->id = 'omise_googlepay';
+        $this->has_fields = true;
+        $this->method_title = __('Omise Google Pay', 'omise');
+        $this->method_description = wp_kses(
+            __('Accept payments through <strong>Google Pay</strong> via Omise payment gateway.', 'omise'),
+            array('strong' => array())
+        );
+
+        $this->supports = array('products', 'refunds');
+
+        $this->init_form_fields();
+        $this->init_settings();
+        $this->register_omise_googlepay_scripts();
+
+        $this->title = $this->get_option('title');
+        $this->description = $this->get_option('description');
+        $this->restricted_countries = array('TH', 'JP', 'SG', 'MY');
+
+        add_action('woocommerce_api_' . $this->id . '_callback', 'Omise_Callback::execute');
+        add_action('woocommerce_update_options_payment_gateways_' . $this->id, array($this, 'process_admin_options'));
+        add_action('wp_enqueue_scripts', array($this, 'omise_scripts'));
+        add_action('woocommerce_order_action_' . $this->id . '_charge_capture', array($this, 'process_capture'));
+        add_action('woocommerce_order_action_' . $this->id . '_sync_payment', array($this, 'sync_payment'));
+    }
+
+    /**
+     * @see WC_Settings_API::init_form_fields()
+     * @see woocommerce/includes/abstracts/abstract-wc-settings-api.php
+     */
+    public function init_form_fields()
+    {
+        $this->form_fields = array(
+            'enabled' => array(
+                'title' => __('Enable/Disable', 'omise'),
+                'type' => 'checkbox',
+                'label' => __('Enable Omise Google Pay Payment', 'omise'),
+                'default' => 'no'
+            ),
+
+            'merchant_id' => array(
+                'title' => __('Google Pay Merchant ID', 'omise'),
+                'type' => 'text',
+                'description' => __('The merchant ID will be available after registering with the <a href="https://pay.google.com/business/console">Google Pay Business Console</a>. (Not needed for test mode)', 'omise')
+            ),
+
+            'request_billing_address' => array(
+                'title' => __('Google Pay Request Billing Address', 'omise'),
+                'type' => 'checkbox',
+                'description' => __('Request customer\'s name and billing address from their Google Account upon checkout.', 'omise'),
+                'default' => 'no'
+            ),
+
+            'request_phone_number' => array(
+                'title' => __('Google Pay Request Phone Number', 'omise'),
```

Comment:
Do we have to add `Google Pay` in front of `Merchant ID`, `Request Billing Address` and `Request Phone Number`? Since we are in Google Pay payment page, I find those self explanatory plus there are helper text below the input fields..

Reply:
yeh let me remove it, good point

Reply:
modified here